lblod/ember-rdfa-editor
# GN-4714: improvements to html-editor

### Overview
This PR contains improvements to the html-editor contained in the `debug-tools` component:
- Removal of obsolete/unused xml-editor features
- Addition of a reusable `codemirror` modifier with support for custom extensions
   * codemirror-editor can also be update 'from the outside': if any of the modifier params change (`content`, `onUpdate` or `extensions`), the codemirror editor is correctly updated, without causing infinite re-validation/loop bugs.
- Improvements to html formatting
   * Replaced `xml-formatter` by `js-beautify`
   * Content located in the following tags is not formatted: `p`, header-tags, `span`, `a`
- Introduction of seperate html-editor modal component
- Addition of toolbar-button which allows opening the html-editor
- Adjustments to styles of toolbar-button component to better support text children

TODO:

- [x] Add seperate component which opens up the `html-editor` modal with improved UI
- [x] (optional) Improve `code-mirror` modifier so it reacts to content changes from the outside

##### connected issues and PRs:
[GN-4714](https://binnenland.atlassian.net/browse/GN-4714?atlOrigin=eyJpIjoiZGVjNDBhZTMwZTE0NDYxOWFjY2RmNmQ4ZWRlNzA3ODUiLCJwIjoiaiJ9)

### How to test/reproduce
- `pnpm start`
- insert any sample data
- open up the html editor
- notice the html is formatted, except for phrasing content between the aforementioned tags
- when setting the html, the spacing should be unchanged
- notice that the `xml` editor option is gone

### Challenges/uncertainties
While working on this feature, I once again noticed a lot of inconsistencies in the styles of our toolbar components. We should probably tackle these in a seperate PR.

### Checks PR readiness
